crates/aggkit-prover/src/aggchain_proof/service.rs

```diff
@@ -0,0 +1,78 @@
+use std::{
+    future::Future,
+    pin::Pin,
+    task::{Context, Poll},
+};
+
+use aggchain_proof_builder::{AggchainProofBuilderService, Request as ProofBuilderRequest};
+use proposer_service::{ProposerService, Request as ProposerRequest};
+
+use super::error::Error;
+
+pub struct ProofRequest {
+    #[allow(dead_code)]
+    pub start_block: u64,
+    #[allow(dead_code)]
+    pub max_block: u64,
+}
+
+pub struct ProofResponse {}
+
+/// The Aggchain proof service is responsible for orchestrating an Aggchain
+/// proof generation.
+///
+/// The Aggchain proof is generated by fetching the Aggregated FEP from the
```

Comment:
I wonder if we should have a separate service for bridge proof generation, or just implement it as a module of the `aggchain-proof-builder`

Reply:
We can if we feel that it makes sense, this PR is just containing a bunch of things to iterate over :)